app/models/export/activity_forecast_columns.rb

```diff
@@ -19,8 +19,7 @@ def headers
 
   def rows
     return [] if @activities.empty?
-
-    @activities.map { |activity|
+    @_rows ||= @activities.map { |activity|
```

Comment:
@ed, the forecasts are already cached pretty heavily, but it is even harder to test this, do you feel the same way about the forecasts class spec?

Reply:
I certainly don't think it's essential!

Looking at `ActivityForecastColumns`, would it be a case of asserting that `ForecastOverview` receives `:new` `1 x n` rather than `3 x n` times?

Is the difficulty in stubbing out the values returned by `ForecastOverview.new(activity_ids).latest_values`?

Reply:
but if the `ForecastOverview` already caches the expensive work (i.e. is safe to call repeatedly) then we shouldn't worry about memoising with `@_rows ||=` -- nor demonstrating that we have done so in a test!

Reply:
looking at 

https://github.com/UKGovernmentBEIS/beis-report-official-development-assistance/blob/master/app/models/export/activity_forecast_columns.rb#L40-L48

I see that the calls to ForecastOverview are themselves memoised, so yes, that looks well and truly covered already!

Reply:
Yep, I think it is stubbing `latest_values` was my issue last night - looking at the whole class in the cold light of day this morning, the caching on rows is not really needed.

I've taken the caching off the rows method in `ActivityForecastColumns` and added coverage around the main caching of `ForecastOverview` as you mentioned - this has been pulled out into a seperate commit.

Would you mind giving it another look for my peace of mind?